src/components/Search/SearchBar/index.tsx

```diff
@@ -66,34 +62,26 @@ const SearchBar: React.FC<SearchBarProps> = ({
     { disabled, ignoreFocus: true }
   );
 
+  const debounceSearch = useRef(
+    debounce((value: string) => {
+      onChange(value);
+    }, 1000)
+  ).current;
+
   const handleOnSubmit = () => {
-    if (inputRef.current) {
-      inputRef.current.blur();
-    }
-    if (isFilter || isHighlight) {
-      setInput("");
-    }
+    // Cancel the debounce request and clear input.
+    debounceSearch.cancel();
+    inputRef.current?.blur();
```

Comment:
There's an interesting side effect of this change. Paginating by pressing enter is no longer as intuitive. Since we no longer blur the input pressing enter no longer paginates the search since the input is still focused. I think we would want to fix that since it is frustrating.

Reply:
Yeah, I noticed that but figured it would be remedied by [EVG-18576](https://jira.mongodb.org/browse/EVG-18576). I ended up just doing the ticket in this PR.

Reply:
Oh also, I realized that `Shift + Enter` wouldn't work for submitting since that is what's used to paginate backwards. I changed it to `Cmd + Shift + Enter`.